File: libsignal/service/src/test/java/org/whispersystems/signalservice/internal/push/GroupStaleDevicesTest.java
package org.whispersystems.signalservice.internal.push;

import org.junit.Test;
import org.whispersystems.signalservice.internal.util.JsonUtil;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class GroupStaleDevicesTest {

  @Test
  public void testSimpleParse() throws IOException {
    GroupStaleDevices[] parsed = JsonUtil.fromJson("[\n" +
        "  {\n" +
        "    \"uuid\": \"12345678-1234-1234-1234-123456789012\",\n" +
        "    \"devices\": {\n" +
        "      \"staleDevices\": [3]\n" +
        "    }\n" +
        "  },\n" +
        "  {\n" +
        "    \"uuid\": \"22345678-1234-1234-1234-123456789012\",\n" +
        "    \"devices\": {\n" +
        "      \"staleDevices\": [2]\n" +
        "    }\n" +
        "  }\n" +
        "]", GroupStaleDevices[].class);

    assertEquals(2,  parsed.length);
    assertEquals("12345678-1234-1234-1234-123456789012", parsed[0].getUuid());
    assertEquals(3, (int) parsed[0].getDevices().getStaleDevices().get(0));

    assertEquals("22345678-1234-1234-1234-123456789012", parsed[1].getUuid());
    assertEquals(2, (int) parsed[1].getDevices().getStaleDevices().get(0));
  }
}

File: lib/libsignal-service/src/test/java/org/whispersystems/signalservice/internal/push/GroupStaleDevicesTest.kt
/*
 * Copyright 2023 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

import assertk.assertThat
import assertk.assertions.containsExactly
import assertk.assertions.hasSize
import assertk.assertions.isEqualTo
import org.junit.Test
import org.whispersystems.signalservice.internal.util.JsonUtil

class GroupStaleDevicesTest {
  @Test
  fun testSimpleParse() {
    val json = """
      [
        {
          "uuid": "12345678-1234-1234-1234-123456789012",
          "devices": {
            "staleDevices": [3]
          }
        },
        {
          "uuid": "22345678-1234-1234-1234-123456789012",
          "devices": {
            "staleDevices": [2]
          }
        }
      ]
    """.trimIndent()

    val parsed: Array<GroupStaleDevices> = JsonUtil.fromJson(json, Array<GroupStaleDevices>::class.java)

    assertThat(parsed).hasSize(2)
    val (first, second) = parsed

    assertThat(first.uuid).isEqualTo("12345678-1234-1234-1234-123456789012")
    assertThat(first.devices.staleDevices).containsExactly(3)

    assertThat(second.uuid).isEqualTo("22345678-1234-1234-1234-123456789012")
    assertThat(second.devices.staleDevices).containsExactly(2)
  }
}
